framework.py

```diff
@@ -185,8 +218,21 @@ def __retrieve_certificates_from_shotgun(self):
         self.logger.debug("Retrieving certificates from Shotgun")
         certs = self.shotgun._call_rpc("sg_desktop_certificates", {})
         sgtk.util.filesystem.ensure_folder_exists(self._get_shotgunlocalhost_keys_folder())
-        self._write_cert("server.crt", certs["sg_desktop_cert"])
-        self._write_cert("server.key", certs["sg_desktop_key"])
+        if not certs["sg_desktop_cert"]:
```

Comment:
Locally hosted sites by default do not have the certs installed, so we should log that error and ask clients to contact support.